src/settings/NoticePolicy/components/EditSections/GeneralSection/GeneralSection.test.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  within,
+} from '@testing-library/react';
+
+import '../../../../../../test/jest/__mock__';
+
+import { Field } from 'react-final-form';
+
+import {
+  TextArea,
+  TextField,
+  Checkbox,
+} from '@folio/stripes/components';
+import { Metadata } from '../../../../components';
+import GeneralSection from './GeneralSection';
+import styles from './GeneralSection.css';
+
+jest.mock('../../../../components', () => ({
+  Metadata: jest.fn(() => null),
+}));
+
+describe('GeneralSection', () => {
+  const generalLabelId = 'ui-circulation.settings.noticePolicy.generalInformation';
+  const nameLabelId = 'ui-circulation.settings.noticePolicy.policyName';
+  const activeLabelId = 'ui-circulation.settings.noticePolicy.active';
+  const descriptionLabelId = 'ui-circulation.settings.noticePolicy.policyDescription';
+  const getById = (id) => within(screen.getByTestId(id));
+  const mockedMetadata = {
+    data: 'testData',
+  };
+  const mockedConnect = jest.fn();
+  const defaultProps = {
+    isOpen: true,
+    metadata: mockedMetadata,
+    connect: mockedConnect,
+  };
+
+  afterEach(() => {
+    Field.mockClear();
+    Metadata.mockClear();
+  });
+
+  describe('when "isOpen" prop is true', () => {
+    beforeEach(() => {
+      render(
+        <GeneralSection
+          {...defaultProps}
+        />
+      );
+    });
+
+    it('should render main label', () => {
+      expect(getById('generalTestId').getByText(generalLabelId)).toBeVisible();
+    });
+
+    it('should pass "isOpen" prop correctly', () => {
+      expect(screen.getByTestId('generalTestId')).toHaveAttribute('open');
+    });
+
+    it('should execute "Metadata" with passed props', () => {
+      const expectedResult = {
+        connect: mockedConnect,
+        metadata: mockedMetadata,
+      };
+
+      expect(Metadata).toHaveBeenLastCalledWith(expectedResult, {});
+    });
+
+    it('should execute "Field" associated with "name" with passed props', () => {
+      const expectedResult = {
+        id: 'notice_policy_name',
+        name: 'name',
+        required: true,
+        autoFocus: true,
+        component: TextField,
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(1, expect.objectContaining(expectedResult), {});
```

Comment:
as discussed to replace index with map in all places

Reply:
done